src/lang/KclSingleton.ts

```diff
@@ -166,8 +172,12 @@ export class KclManager {
     this.engineCommandManager = engineCommandManager
 
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.ensureWasmInit().then(() => {
-      this.ast = this.safeParse(codeManager.code) || this.ast
+    this.ensureWasmInit().then(async () => {
+      void this.safeParse(codeManager.code).then((ast) => {
```

Comment:
i did this dirty too @jtran

Reply:
For this line, `void` can actually change to `await` since the closure is async.

But `ensureWasmInit()` is still dirty. I'll have to think about this one. I don't think it's worth changing in this PR.